backend/app/core/auth.py

```
@@ -1,177 +1,133 @@
+            if not self.password_service.verify_password(password, user.hashed_password):
+                return None
 
-        return user
+            # Check if password hash needs updating
+            new_hash = self.password_service.update_hash_if_needed(password, user.hashed_password)
+            if new_hash:
+                user.hashed_password = new_hash
+                await self.db.commit()
+
+            return user
+        except Exception:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)